### PR TITLE
8294438: Fix misleading-indentation warnings in hotspot

### PR DESCRIPTION
**Pull request by shipilev**
There are number of places where misleading-indentation is reported by GCC. Currently, the warning is disabled for the entirety of Hotspot, which is not good.

C1 does an unusual style here. Changing it globally would touch a lot of lines. Instead of doing that, I fit the existing style while also resolving the warnings. Note this actually solves a bug in `lir_alloc_array`, where `do_temp` are called without a check.

Build-tested this with product of:
  - GCC 10
  - {i686, x86_64, aarch64, powerpc64le, s390x, armhf, riscv64}
  - {server, zero}
  - {release, fastdebug}

Linux x86_64 fastdebug `tier1` is fine.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294438](https://bugs.openjdk.org/browse/JDK-8294438): Fix misleading-indentation warnings in hotspot

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to [0631c74f](https://git.openjdk.org/jdk/pull/10444/files/0631c74f55ead1733a0db0f050b19e1641cbe1f8)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [0631c74f](https://git.openjdk.org/jdk/pull/10444/files/0631c74f55ead1733a0db0f050b19e1641cbe1f8)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [0631c74f](https://git.openjdk.org/jdk/pull/10444/files/0631c74f55ead1733a0db0f050b19e1641cbe1f8)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10444/head:pull/10444` \
`$ git checkout pull/10444`

Update a local copy of the PR: \
`$ git checkout pull/10444` \
`$ git pull https://git.openjdk.org/jdk pull/10444/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10444`

View PR using the GUI difftool: \
`$ git pr show -t 10444`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10444.diff">https://git.openjdk.org/jdk/pull/10444.diff</a>

</details>

### CODE REVIEW

**Comment by magicus** [comment]
So enabling the warning actually exposed a bug? That's always nice to see, and a good argument for keeping up with the hygiene work of minimizing the amount of disabled warnings.

**Comment by shipilev** [comment]
> So enabling the warning actually exposed a bug? That's always nice to see, and a good argument for keeping up with the hygiene work of minimizing the amount of disabled warnings.

Yes. It is hard to judge how innocuous that bug really is, because (handwaves) compilers.

**Review by magicus — APPROVED** [review]
The build part of this PR is OK

**Review by dholmes-ora — APPROVED** [review]
Changes look good!

That C1 code is very strangely written!

Thanks

File: src/hotspot/share/classfile/javaClasses.cpp [hunk 1]
```diff
@@ -3923,7 +3923,9 @@ oop java_lang_boxing_object::initialize_and_allocate(BasicType type, TRAPS) {
   Klass* k = vmClasses::box_klass(type);
   if (k == NULL)  return NULL;
   InstanceKlass* ik = InstanceKlass::cast(k);
-  if (!ik->is_initialized())  ik->initialize(CHECK_NULL);
```
[the remark this answers is not in the document]
Comment (shipilev) [reply src/hotspot/share/classfile/javaClasses.cpp:3926]: How is this a bu... ah. Oh! The second statement from `CHECK_NULL` is effectively out of `if (!ik->is_initialized())` block!

[the remark this answers is not in the document]
Comment (dholmes-ora) [reply src/hotspot/share/classfile/javaClasses.cpp:3926]: Yep another CHECK macro gotcha!

[the remark this answers is not in the document]
Comment (dholmes-ora) [reply src/hotspot/share/classfile/javaClasses.cpp:3926]: Harmless in this case as no pre-existing exception should be pending.

**Comment by shipilev** [comment]
Thanks!

/integrate

**Comment by dholmes-ora** [comment]
@shipilev  this has broken our linux aarch64 builds!

[2022-10-20T07:26:59,542Z] workspace/open/src/hotspot/cpu/aarch64/assembler_aarch64.cpp: In member function 'void Address::lea(MacroAssembler*, Register) const':
[2022-10-20T07:26:59,542Z] workspace/open/src/hotspot/cpu/aarch64/assembler_aarch64.cpp:138:5: error: this 'else' clause does not guard... [-Werror=misleading-indentation]
[2022-10-20T07:26:59,542Z]   138 |     else
[2022-10-20T07:26:59,542Z]       |     ^~~~
[2022-10-20T07:26:59,542Z] workspace/open/src/hotspot/cpu/aarch64/assembler_aarch64.cpp:140:7: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'else'
[2022-10-20T07:26:59,542Z]   140 |       break;
[2022-10-20T07:26:59,542Z]       |       ^~~~~

**Comment by shipilev** [comment]
> @shipilev this has broken our linux aarch64 builds!

Whoa. Looking.

**Comment by shipilev** [comment]
> > @shipilev this has broken our linux aarch64 builds!
> 
> Whoa. Looking.

That would be: #10781

**Pull request closed without merging**